libyul/backends/wasm/BinaryTransform.cpp

```diff
@@ -556,16 +607,22 @@ bytes BinaryTransform::memorySection()
 	return makeSection(Section::MEMORY, std::move(result));
 }
 
-bytes BinaryTransform::globalSection()
+bytes BinaryTransform::globalSection(vector<GlobalVariableDeclaration> const& _globals)
 {
+	yulAssert(m_globals.size() == _globals.size(), "");
+	// ASSUMPTION: _globals[i] and m_globals[i] are the same variable
```

Comment:
Maybe there's an overhaul due here since `functionSection` takes `_function` but also there's `m_function` with some relevant content (can be done in a separate PR).

Reply:
@axic I tried refactoring it a bit. Please take a look at #9151.